test functions for make_sphere.jl

LongKhanhDuong opened the pull request:
added test functions for make_sphere.jl in runtests.jl

removed times_two() from the codebase

added input validation for error handling in make_sphere.jl.

I need guidance on how to effectively test the noise

702ph reviewed (changes requested):
each one should have one pull request. there seems to be three changes in different areas.

LongKhanhDuong commented:
My changes are all related to the `make_sphere.jl` test. The original code lacked input validation, so I added that. I only wrote  tests for `make_sphere` and removed the `times_two` function, which we had to remove anyway.

702ph commented:
If the changes are dependent on each other, they should be in the same pull request. However, if they are only related but not dependent, they should be submitted separately. In other words, If the test methods can work without the changes to [this line](https://github.com/702ph/SpectralClusteringTools.jl/blob/465f11d178d6f999df86fe77e27a96acc152141d/src/Utils/DataSets.jl#L23), then those changes should be moved to a separate pull request.

Separating unrelated changes makes the review process easier and helps prevent unwanted or incorrect merges caused by misunderstandings. It also reduces workload by keeping the changes less complex and more focused, which improves maintainability and clarity in the project's history.

702ph commented:
In above case, for a separate branch and pull request, I suggest using "enhancement" or "improvement" as a branch name prefix.

702ph commented:
I closed the pull request.
